pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -639,16 +645,14 @@ private static String database(Properties props) {
   /**
    * @return the timeout from the URL, in milliseconds
    */
-  private static long timeout(Properties props) {
-    String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
-    if (timeout != null) {
-      try {
-        return (long) (Float.parseFloat(timeout) * 1000);
-      } catch (NumberFormatException e) {
-        LOGGER.log(Level.WARNING, "Couldn't parse loginTimeout value: {0}", timeout);
-      }
+  private static long timeout(Properties props) throws PSQLException {
+    long result = 0L;
+    if (PGProperty.LOGIN_TIMEOUT.isPresent(props)) {
+      result = (long) (PGProperty.LOGIN_TIMEOUT.getFloat(props) * 1000);
+    } else {
+      result = (long) (DriverManager.getLoginTimeout() * 1000);
```

Comment:
I think the proper code would be to express `1000` as a long as in `1000L` otherwise there's a hidden silent integer overflow: https://stackoverflow.com/a/7215432/1261287